src/AutoEnvMap.php

```diff
@@ -105,14 +109,43 @@ private function consumeParameter($param, $default, $envPrefix = '')
     {
         $envName = $this->paramToEnvName($param, $envPrefix);
 
+        // ENV is set for this parameter
         if (getenv($envName) !== false) {
-            $this->envMap[$param] = $envName;
-        } elseif (is_array($default)) {
+            return $envName;
+        }
+
+        // Support nested parameters and flatten for ParameterHandler
+        if (is_array($default)) {
+            $canBuild = true;
+            $build = array();
+
             foreach ($default as $subParam => $subDefault) {
-                $this->consumeParameter($param . '.' . $subParam, $subDefault, $envPrefix);
+                $subEnvName = $this->consumeParameter($param . '.' . $subParam, $subDefault, $envPrefix);
+
+                // If any sub-parameters are missing we can't build, but continue in order to log all of them
+                if ($subEnvName === '') {
+                    $canBuild = false;
+                }
+
+                $build[] = $subParam . ': ' . getenv($subEnvName);
             }
-        } else {
-            $this->missingParameters[$param] = $envName;
+
+            if (!$canBuild) {
+                // Return without alerting that the top-object as missing
+                return '';
+            }
+
+            // Build yaml object (all env values should be valid yaml)
+            $envValue = '{' . implode(', ', $build) . '}';
+
+            putenv($envName . '=' . $envValue);
+
+            return $envName;
         }
+
+        // If we've got this far then the ENV for this parameter is missing / incomplete
+        $this->missingParameters[$param] = $envName;
```

Comment:
Instead of this being at the bottom, your code would be more legible if you placed this in a `if (!is_array($default)) {` block, and have the main functionality of the method _outside_ of an `if` block.